### PR TITLE
Added support for Android N

### PR DESCRIPTION
**Pull request by hdezninirola**
Android removed the option of running native with text relocations: https://android.googlesource.com/platform/bionic/+/e4ad91f86a47b39612e030a162f4793cb3421d31%5E%21/

This commit simply bumps FFmpeg from 2.4 to 3.0, starts using NDK revision 12b, and fixes a few things that changed in different repos and should allow building again. It also includes a fix for #3.

Also added script to install Auto-tools.

### CODE REVIEW

File: autotools.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+curl -OL http://ftpmirror.gnu.org/autoconf/autoconf-2.69.tar.gz
```
Comment (coltin) [inline autotools.sh:4]: This file could us a brief comment describing what it does.

Also #!/bin/bash at the top

File: build_ffmpeg.sh [hunk 2]
```diff
@@ -32,22 +32,22 @@ if [ "$1" = "--init" ]; then
...
+        printf "    downloading r12b for mac\n"
+        curl "https://dl.google.com/android/repository/android-ndk-r12b-darwin-x86_64.zip" \
+            -o android-ndk-r12b-darwin-x86_64.zip \
             >> ${LOG_FILE} 2>&1
     fi
 
-    printf "    extracting\n"
-    tar xvzf android-ndk32-r10-darwin-x86_64.tar.bz2 \
+    printf "    unzipping\n"
+    unzip android-ndk-r12b-darwin-x86_64.zip \
```
Comment (coltin) [inline build_ffmpeg.sh:43]: nit: Could make ndk-r12b into a constant so it's easy to update in the future. Feel free to merge without this change.

File: build_ffmpeg.sh [hunk 3]
```diff
@@ -58,11 +58,11 @@ if [ "$1" = "--init" ]; then
     cd ${DIR_NDK}/sources \
         >> ${LOG_FILE} 2>&1
 
-    printf "    downloading yasm\n"
+    printf "    cloning yasm\n"
```
Comment (coltin) [inline build_ffmpeg.sh:61]: +1

**Comment by coltin** [comment]
Made a few minor comments, feel free to merge! SHIP IT.

File: build_ffmpeg.sh (hunk 2 above)
Comment (hdezninirola) [reply build_ffmpeg.sh:43]: They might change it in a different pattern next time though. In r11 they changed it from tar.bz2 to zip for example, I think we can leave it as is.

**Pull request merged**